File: src/receiver.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void callback(const std_msgs::String& msg) {
    ROS_INFO("Message received: %s", msg.data.c_str());
}

int main (int argc, char **argv) {
    ros::init(argc, argv, "receiver");
    
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("/test_message", 1000, callback);

    ros::spin(); // keep node running until it is shut down
}

File: scripts/receiver.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import String

def callback(msg):
    rospy.loginfo("Message received: {}".format(msg))

if __name__ == '__main__':
    rospy.init_node("receiver")

    sub = rospy.Subscriber("/test_message", String, callback)

    rospy.spin() # will keep script running until node is stopped